nilearn/datasets/utils.py

```diff
@@ -310,6 +310,24 @@ def _get_dataset_dir(dataset_name, data_dir=None, default_paths=None,
                   'directories, but:' + ''.join(errors))
 
 
+def _is_within_directory(directory, target):
+    abs_directory = os.path.abspath(directory)
+    abs_target = os.path.abspath(target)
+
+    prefix = os.path.commonprefix([abs_directory, abs_target])
+
+    return prefix == abs_directory
+
+
+def _safe_extract(tar, path=".", members=None, *, numeric_owner=False):
```

Comment:
could you add a link to the original issue and https://github.com/advisories/GHSA-gw9q-c7gh-j9vm to keep track of where this came from? thanks! LGTM otherwise

Reply:
Done!